1Password/pulumi-onepassword
# Add E2E testing for NodeJS example

## Summary
This PR adds the necessary code to have end-to-end testing for the NodeJS provider example present in the codebase.

This was done by following the [Add End-to-end Testing](https://github.com/pulumi/pulumi-tf-provider-boilerplate?tab=readme-ov-file#add-end-to-end-testing) instructions.

## How to test

Before running the test, make sure you have installed the tools mentioned in [this list of requirements](https://github.com/pulumi/pulumi-tf-provider-boilerplate?tab=readme-ov-file#prerequisites).

1. Export the `OP_SERVICE_ACCOUNT_TOKEN` environment variable:
   ```sh
   export OP_SERVICE_ACCOUNT_TOKEN=<your-service-account-token>
   ```

2. Get an item that you want to fetch for this testing and extract the item and vault IDs:
   ```sh
   op item get --vault <your-vault> <your-item>
   ```

1. Export the `OP_VAULT_UUID` and `OP_ITEM_UUID` to the values obtained from previous step:
   ```sh
   export OP_VAULT_UUID=<your-vault-uuid> && \
       export OP_ITEM_UUID=<your-item-uuid>
   ```

1. Build the provider binary:

   ```sh
   make provider
   ```

3. In the terminal session, temporarily include the `bin` directory created to your PATH:

   ```sh
   export PATH=$PWD/bin:$PATH
   ```

4. Build the Node SDK:

   ```sh
   make build_nodejs
   ```

5. Link the local Node SDK to be used in testing:

   ```sh
   make install_nodejs_sdk
   ``` 

6. Run the testing command:

   ```sh
   make test
   ```
   - [x] The test should succeed with no issues.
